services/explorer-indexer/src/modules/live-sync/live.syncer.service.ts

```diff
@@ -44,7 +52,23 @@ export default class LiveSyncerService {
     this.logger.log(
       `Connected to WS server at ${this.ws.url}. Blockchain sync started.`
     );
-
+    const storedGenesis = await this.pbftService.getBlockByNumber(0);
+    const chainGenesis = (
+      await this.graphQLConnector.getPBFTBlocksByNumberFromTo(0, 0)
+    )[0];
+    if (
+      storedGenesis &&
+      chainGenesis &&
+      storedGenesis.hash.toLowerCase() !== chainGenesis.hash.toLowerCase()
+    ) {
+      this.logger.warn('New genesis block hash detected. Wiping database.');
+      await this.txService.clearTransactionData();
+      this.logger.warn('Cleared Transaction table');
+      await this.dagService.clearDagData();
+      this.logger.warn('Cleared DAG table');
+      await this.pbftService.clearPbftData();
+      this.logger.warn('Cleared PBFT table');
+    }
```

Comment:
Let`s also clear the queue just in case:
    ```
 await this.pbftsQueue.empty();
      this.logger.warn('Cleared PBFT queue');
      await this.dagsQueue.empty();
      this.logger.warn('Cleared DAG queue');
```